server/middlewares/index.js

```diff
@@ -1,13 +1,45 @@
-const {decodeToken} = require('../utils');
+const { decodeToken } = require('../utils');
+const {User} = require('../models');
 
-exports.verifyJwt = (req, res, next) => {
-    const {authorization} = req.headers;
-    if (authorization) {
-        const token = authorization.split(' ')[1];
-        const decodedToken = decodeToken(token);
-        req.user = decodedToken.payload;
-        next();
+exports.isAuth = (req, res, next) => {
+  const { authorization } = req.headers;
+  if (authorization) {
+    const token = authorization.split(' ')[1];
+    const { payload } = decodeToken(token);
+    if (payload && payload._id) {
+      User.findById(payload._id)
+        .populate('projects')
+        .exec((err, user) => {
+          if (err || !user) {
+            res.status(400).json({
+              error: 'User not found'
+            });
+          } else {
+            const { _id, name, email, projects } = user;
+            req.profile = { _id, name, email, projects };
+            next();
+          }
+        });
     } else {
-        return res.status(500).send({error: 'not authorized'})
+      return res.status(403).send({ error: 'Access denied' });
     }
-}
\ No newline at end of file
+  } else {
+    return res.status(403).send({ error: 'Access denied' });
+  }
+};
+
+//IS THIS NEEDED BY FUTURE ROUTES?
```

Comment:
I actually thought about this and think it would be best to keep the userById param method. In this `.userById()` middleware method, it can be used to get any user via the params `:id` when used in router.param('userId', userById). It gets the user associated with the :userId and then sets them to a .profile property on the req object. 
So if we were to do a get user like router.use(''users:/userId', getUser) the getUser controller can simply return the req.user
So anytime there is a :userId in our route, the user information will be set to req.user. Works for the user logged in or looking up any other user as well.

Reply:
It's definitely useful. My question was whether it was needed if the isAuth middleware was used too. For the branches that existed, it doesn't. However, I don't want the verification to do too much and think it should probably be separated into uses so that it is easier to build with. I've been changing it up this morn.

Reply:
It isn't necessarily necessary in isAuth but I did use it for comparisons to compare the id on the token to the id from the params. At the moment if you made a request to create a project on projects/add/:userId or projects/add/:someOtherUsersId you would be able to create a project associated with the user in the jwtToken. Nothing breaking though but in the earlier version with userById it would fail if you were to make a request on /add/:someOtherUsersId with the way it was setup before because the user on req.profile from userById (it would fetch a different users profile) and compared it to the jwtToken payload (which would be the logged in user). 
It doesn't matter in this case since what matters most is that the user is only allowed to create projects for themselves. And it's not possible in either way (as far as I can tell). 
It can also be used to populate the users' projects as well. So whenever a route includes `:userId` it will have user and their projects on req.profile. I'm planning to use it for getting all projects probably in a route like '/users/:userId/projects' to exclude projects by userId

Reply:
I absolutely understand how it is mean to work -- it is very standard and simple. I was considering uses in our app and as I discussed with Evan in regards to his route which utilized it, I would not mess with it without making sure everything was good and settled.

Reply:
Okay, I think I was misunderstanding the commenting out. I was under the impression you were going to remove it if it wasn't being used but it will definitely be used in the future. Sorry about that

Reply:
(I'm going to deal with the middlewares in a different branch)

Reply:
Nothing to be sorry about. I put too much in this branch and it muddled everything.